libs/MeshKernel/tests/src/PolygonTests.cpp

```diff
@@ -175,3 +175,101 @@ TEST(PolygonTests, FailureConstructionTests)
     std::vector<mk::Point> closedPolygonWithNull{{-10.0, -5.0}, {20.0, -5.0}, {20.0, 10.0}, {-10.0, 10.0}, {mk::constants::missing::doubleValue, mk::constants::missing::doubleValue}, {-10.0, -5.0}};
     EXPECT_THROW([[maybe_unused]] mk::Polygon polygon(closedPolygonWithNull, mk::Projection::cartesian), mk::ConstraintError);
 }
+
+void CheckPolygonPointVectors(const std::vector<mk::Point>& actual, const std::vector<mk::Point>& expected)
+{
+    ASSERT_EQ(actual.size(), expected.size());
+    for (size_t i = 0; i < actual.size(); ++i)
+    {
+        constexpr double tolerance = 1e-6;
+        EXPECT_NEAR(actual[i].x, expected[i].x, tolerance);
+        EXPECT_NEAR(actual[i].y, expected[i].y, tolerance);
+    }
+}
+
+TEST(PolygonTests, Refine_WhenStartIndexLessThanEndIndex_ThenRefinesBetweenStartIndexAndEndIndex)
+{
+    // setup
+    const std::vector<mk::Point> outer{{0., 0.}, {5., 0.}, {5., 5.}, {0, 5.}, {0., 0.}};
+    const std::vector<mk::Point> expected{{0., 0.}, {5., 0.}, {5., 2}, {5., 4}, {5., 5.}, {3, 5}, {1, 5}, {0, 5.}, {0., 0.}};
+    const mk::Polygon polygon(outer, mk::Projection::cartesian);
+
+    // call
+    const auto refined = polygon.Refine(1, 3, 2);
+
+    // assert
+    CheckPolygonPointVectors(refined, expected);
```

Comment:
out of curiosity, if this fails, what line number does gtest show in the log? one from the body of CheckPolygonPointVectors?

Reply:
https://google.github.io/googletest/advanced.html#using-assertions-in-sub-routines

I added SCOPED_TRACE calls.